pallets/schemas/src/lib.rs

```diff
@@ -560,57 +846,148 @@ pub mod pallet {
 			<CurrentSchemaIdentifierMaximum<T>>::set(n);
 		}
 
+		/// Set the schema count to something in particular.
+		#[cfg(any(feature = "std", feature = "runtime-benchmarks", test))]
+		pub fn set_intent_count(n: IntentId) {
+			<CurrentIntentIdentifierMaximum<T>>::set(n);
+		}
+
+		/// Lowest-level insertion function for a [`SchemaInfo`] and [`SchemaPayload`] into storage,
+		/// using an already-allocated [`SchemaId`]
+		pub fn store_schema_info_and_payload(
+			schema_id: SchemaId,
+			schema_info: SchemaInfo,
+			schema_payload: BoundedVec<u8, T::SchemaModelMaxBytesBoundedVecLimit>,
+		) -> Result<(), DispatchError> {
+			<SchemaInfos<T>>::insert(schema_id, schema_info);
+			<SchemaPayloads<T>>::insert(schema_id, schema_payload);
+			Ok(())
+		}
+
 		/// Inserts both the [`SchemaInfo`] and Schema Payload into storage
 		/// Updates the `CurrentSchemaIdentifierMaximum` storage
 		pub fn add_schema(
+			intent_id: IntentId,
 			model: BoundedVec<u8, T::SchemaModelMaxBytesBoundedVecLimit>,
 			model_type: ModelType,
 			payload_location: PayloadLocation,
-			settings: BoundedVec<SchemaSetting, T::MaxSchemaSettingsPerSchema>,
-			schema_name_option: Option<SchemaName>,
+			settings: IntentSettings,
+			status: SchemaStatus,
 		) -> Result<SchemaId, DispatchError> {
 			let schema_id = Self::get_next_schema_id()?;
-			let has_name = schema_name_option.is_some();
```

Comment:
question: So what happened with this field that we were storing in SchemaInfo? Are we enforcing all the schemas to be named ones and if so what are we going to do with the existing ones on chain? I'm assuming that might be part of migration

Reply:
Schemas can no longer have names; only `Intents` and `IntentGroups` are named. Unlike with Schemas, where the name was optional, Intent/Group names are mandatory. A Schema implements one and only one _named_ Intent.

The migration ensures that for every existing Schema, we create an equivalent Intent with the same numeric ID, and the identical name (if the Schema had a name).

Q: Theoretically, if the original Schema was un-named, the resulting Intent will also not have a name. This is not necessarily problematic (and not an issue on mainnet where all schemas are named), but if we want to enforce (as the extrinsic does) that ALL Intents are named, we would have to either auto-create a name, or drop any unnamed schemas from Testnet instead of converting them.